File: src/main/java/com/yogesh/meditrack/service/PrescriptionService.java
package com.yogesh.meditrack.service;

import com.yogesh.meditrack.entity.Prescription;
import java.util.List;

public interface PrescriptionService {

    Prescription savePrescription(Prescription prescription);

    List<Prescription> getAllPrescriptions();

    Prescription getPrescriptionById(Long id);

    Prescription updatePrescription(Long id,
                                    Prescription prescription);

    void deletePrescription(Long id);

}

File: frontend/src/service/prescriptionService.js
import api from "./api";

export const getAllPrescriptions = () => api.get("/prescriptions");

export const getPrescriptionById = (id) => api.get(`/prescriptions/${id}`);

export const savePrescription = (prescription) =>
  api.post("/prescriptions", prescription);

export const updatePrescription = (id, prescription) =>
  api.put(`/prescriptions/${id}`, prescription);

export const deletePrescription = (id) => api.delete(`/prescriptions/${id}`);

export const downloadPrescriptionPdf = (id) =>
  api.get(`/prescriptions/pdf/${id}`, { responseType: "blob" });